pkg/gardenlet/controller/controllerinstallation/controllerinstallation_control.go

```diff
@@ -332,15 +333,16 @@ func (r *reconciler) delete(ctx context.Context, gardenClient client.Client, con
 	if err == nil || apierrors.IsConflict(err) {
 		message := fmt.Sprintf("Deletion of Namespace %q is still pending.", namespace.Name)
 		conditionInstalled = gardencorev1beta1helper.UpdatedCondition(conditionInstalled, gardencorev1beta1.ConditionFalse, "DeletionPending", message)
-		return errors.New(message)
+		log.Info(message)
+		return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
```

Comment:
Nit: Isn't 30 seconds a bit long, how about reducing it to e.g. 10 seconds? Also above for the deletion of managed resources.

Reply:
In general this RequeueAfter approach has the drawback that it can get busy the control loop infinitely if the desired condition is not met (for example the Namespace deletion cannot complete because KCM is down; or any other reason that block/prevents the deletion). The old approach is reusing the underlying exponential backoff retry mechanism and doesn't get the control loop constantly busy when the desired condition is not met.
I can agree that for the general case using RequeueAfter approach can be useful because if for example the ManagedResource/Namespace deletion hangs for 1minute, exponential backoff will fail for several retries and the next retry can potential occur in a later point. So having in mind the case where the desired condition is not met for a much longer time, I would rather vote to do not decrease the RequeueAfter period.